java/src/jmri/jmrit/display/layoutEditor/configurexml/LayoutTurnoutXml.java

```diff
@@ -43,7 +41,7 @@ public Element store(Object o) {
 
         // include attributes
         element.setAttribute("ident", p.getName());
-        element.setAttribute("type", tTypeEnumMap.outputFromEnum(p.getTurnoutType()));
+        element.setAttribute("type", p.getTurnoutType().name());
```

Comment:
Put the mapping back please.  It can be a null map, that's fine, but you need it to go the other direction.

Reply:
The map code is fragile to the ordinal. When we can remove the xml int code we can remove all the ENUM(int) code at that time and switch over to the map. I see no value added and considerable risk switching now.

Reply:
> The map code is fragile to the ordinal. When we can remove the xml int code we can remove all the ENUM(int) code at that time and switch over to the map. I see no value added and considerable risk switching now.

I agree with Bob on this.  The map needs to be put back.

You say it is too fragile, but the fact is the ordinal values are being phased out here in favor of the ordinal, so code for the future solution, not the past solution.

Reply:
NO: the ordinal values are NOT being phases out here in favor of the ordinal: they're being phased out here in favor of the ENUM names.

Reply:
In addition, tTypeEnumMap.outputFromEnum outputs the ordinal; my code saves it by name (with appropriately schema modifications); the map also parses by the ordinal (int). Where as my code "does the right thing" and parses by the ENUM name first; if that falls then it then tries by the ENUM int value. This is the correct thing to do.

Reply:
EnumIO is an incorrect implementation; it saves by ordinal and loads by ordinal… just like the pre-enum code… what exactly is it's "value added"? (because I'm not seeing it.)

Reply:
So basically you want me to remove all my code (and the schema modifications) that stores and loads by enum name so we can use EnumIO and save it by ordinal? Why?

Reply:
EnumIO use restored